### DISCUSSION TITLE
Support console platforms without requiring going through a 3rd party company, remove conflicts of interest

### DISCUSSION THREAD

**Discussion by wjl** [Redot-Engine/redot-proposals#55] [Ideas]
Redot should support self-publishing on console platforms for users who have vendor SDKs, and avoid any kind of conflict of interest with commercial game porting companies.

Apologies if this has been mentioned before, but I did not find anything in either the open issues or discussions that had the word "company" or "W4" in it. =)

On big problem I've had with Godot is that they don't support self-publishing on console platforms. Godot's web-site specifically says:

> Partner with a publisher to target consoles
> 
> If you want to release to a console, you can find several 3rd party publishers which specialize on that.

This sounds reasonable at first glace. Sure, it's great that there are companies that you could hire to help you port a game! I can do that now with Unity, Unreal, etc.

But this gets a little more muddied when you start to see that, the *only* supported way to do a console port is to hire W4 Games, which "just happens" to be run by some of the main Godot developers. That's a weird conflict of interest, but maybe not the end of the world on its own.

What's worse, however, if you ever ask about self-publishing a game on a console platform in their discussion forums, you get preached to about how (1) it's impossible to support commercial SDKs in an open-source project and (2) it's so hard anyway, you shouldn't even try, and you're bad for even wanting to.

This is wrong-headed. For argument #1, although obviously closed source code can't be built directly into the engine, support for 3rd party closed source console vendor SDKs is completely possible, if Redot was willing to add the right hooks and expose the right APIs, etc to make porting clean and easy. Furthermore, these ports could be as open as possible, and shared freely with the subset of the community that has vendor SDK access. For example, Defold does this incredibly well.

For argument #2, it's even worse, because the argument boils down to: it's very, very hard to port Godot to a console. Therefore you should hire W4 Games to port it for you. If it were easier to port Godot to a console platform, then you wouldn't need to hire W4 Games, which would be bad for that company. That company is founded and run by the same people who decide if Godot should modified to be easier to port to consoles. Even if this isn't actually affecting the course of development, this is a *massive* conflict of interest, and shouldn't be tolerated by the community.

I don't have all the answers, but I would propose that Redot explicitly adds some goals or plans to:

1.  Enhance the underlying platform support and API of the Redot Engine to make it easy to port to any arbitrary platform, whether is be existing consoles or anything new that comes out.
2. Commit to *supporting* those in the community who wish to port their games to any platform, self-publish, or pay a company to help them do ports, without any prejudice.
3. Insofar as it is possible, work with console vendors to provide a pre-ported version of Redot to licensees of the vendor SDKs -- this is obviously feasible, as Defold has managed this.

Obviously #3 is only partially open source, and so shouldn't be a high-priority for the project, nor a place where donations, etc should be spent. But it shouldn't be discouraged or actively prevented, as this is harmful to users and developers, and ultimately to acceptance and adpotion of the Redot game engine itself.

**Comment by Arctis-Fireblight** [comment]
We are interested in pursuing a model for this similar to Defold's, but will likely be a very long time before anything like this materializes.
